### DISCUSSION TITLE
Creating a multinetwork test case

### DISCUSSION THREAD

**Discussion by LQ56** [lanl-ansi/PowerModelsDistribution.jl#419] [General]
Hello,

I am trying to build a small multi network test-case by incorporating time series data from the loadshapes.dss file into another test case, but I can't seem to succeed, even with the help of the online documentation.Whenever I run make_multinetwork, PMD tells me that it can't infer time_elapsed, and it only builds network 0. Here is a screenshot:

![Capture_PMD_Multinetwork](https://user-images.githubusercontent.com/115047930/210474966-376d8c8e-ddc6-4816-8848-ca8e5de2e127.PNG)

I am pretty sure I must be doing something wrong but I don't know where,

**Comment by frederikgeth** [comment]
Here's an example that hopefully works for you. Make the following DSS file, that embeds 48 time steps:
```
Clear
New Circuit.N  pu=1.0 basekV=0.415

New Linecode.oh_mars nphases=3  Units=km
~ Rmatrix=[0.54277  |0.103403  0.550964  |0.113221  0.11816  0.573739  ]
~ Xmatrix=[0.61961  |0.297325  0.602324  |0.221825  0.246012  0.555584  ]
~ Cmatrix=[8.80036  |-2.74386  9.28664  |-1.11293  -1.96824  9.28664  ]

New Line.1 Units=km Length=0.010 bus1=sourcebus.1.2.3 bus2=2.1.2.3 switch=n enabled=y phases=3 Linecode=OH_Mars
New Loadshape.LoadProfileP1 npts=48 interval=0.5 mult=[1.697 0.983 0.926 0.914 0.951 0.926 1.012 0.941 0.865 0.975 0.963 0.886 1.015 0.958 1.039 0.928 0.913 2.602 2.325 2.898 2.607 2.410 2.331 2.269 2.523 2.474 2.603 2.208 2.502 2.151 2.121 2.072 2.136 2.595 2.359 2.352 2.421 2.385 2.805 2.934 2.729 2.198 2.250 2.131 2.172 2.189 2.058 1.043]
New Load.LoadP1 bus1=2.1.0 kV=0.24 Vminpu=0.1 Vmaxpu=1.9 model=1 kW=1 pf=0.9 Phases=1 daily=LoadProfileP1
```
and then run in Julia
```
using PowerModelsDistribution
d = parse_file("example.dss", multinetwork=true)
```
This reads in data with 48 time steps:
```
julia> length(d["nw"])
48
```

Does this get you unstuck? Please don't hesitate to come back with questions.

**Comment by LQ56** [comment]
Yes! Thank you very much.

**Comment by pseudocubic** [comment]
Looking at your example, there are a couple of things that are preventing this from working the way you intended. 

@frederikgeth's method is probably the easiest, i.e., defining the loadshape directly in the DSS files, adding `daily=load_shape_name` to the loads you wish to add loadshapes to, and parsing it that way, but if you want to load a LoadShape from somewhere else and apply it to a network like you were trying to do you can do it in the following way:

First, the time series that you load needs to be added to the `eng` data into the top-level `"time_series"` Dict like so (assuming you have already loaded `ts` and `eng` as you did in your example):

```julia
eng["time_series"] = ts["time_series"]
```

Second, instead of adding the raw time series data to the load objects themselves, what you should do is just add "pointers" to the relevant time series data, like so:

```julia
eng["load"]["load1"]["time_series"] = Dict{String,Any}("pd_nom"=>"6", "qd_nom"=>"6")
eng["load"]["load2"]["time_series"] = Dict{String,Any}("pd_nom"=>"6", "qd_nom"=>"6")
```

where the values `pd_nom` and `qd_nom` are the properties that are altered by the time series data and `6` is the name of the `time_series` data that you want to use.
